ubyssey/static/src/js/notifications.js

```diff
@@ -0,0 +1,89 @@
+import DispatchAPI from './api/dispatch';
+import Cookies from 'js-cookie'
+
+const applicationServerPublicKey = 'BOValqIGLJdBIl-qoqRvhPKIa4YxQYDBJsrTCdi7JTVKXGcMpPi6lI26M7s2nteLhAj8zBjDqi40B_vl7Re_iyE';
+
+let isSubscribed = false;
+let swRegistration = null;
+
+function urlB64ToUint8Array(base64String) {
+  const padding = '='.repeat((4 - base64String.length % 4) % 4);
+  const base64 = (base64String + padding)
+    .replace(/\-/g, '+')
+    .replace(/_/g, '/');
+
+  const rawData = window.atob(base64);
+  const outputArray = new Uint8Array(rawData.length);
+
+  for (let i = 0; i < rawData.length; ++i) {
+    outputArray[i] = rawData.charCodeAt(i);
+  }
+  return outputArray;
+}
+
+const cookieName = 'notification_subscription'
+
+function getCookie(field) {
+  let cookie = Cookies.get(cookieName)
+  if (typeof cookie === 'string' && cookie !== '') {
+    cookie = JSON.parse(cookie)
+    if (field) {
+      return cookie[field]
+    }
+    return cookie
+  }
+  return cookie
+}
+
+function setCookie(uuid) {
+  Cookies.set(
+    cookieName,
+    {uuid: uuid},
+    { path: '/' }
+  )
+}
+
```

Comment:
we use these cookie functions in several places, we should make them generalized and move them into a cookies/helpers.js